cilium-cli/clustermesh/clustermesh.go

```diff
@@ -1594,52 +1596,97 @@ func (k *K8sClusterMesh) validateCAMatch(aiLocal, aiRemote *accessInformation) (
 // (certgen) mode. As with classic mode, only autodetected IP-based
 // clustermesh-apiserver Service endpoints are currently supported.
 func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
-	localRelease, err := getRelease(k.client.(*k8s.Client), k.params)
+	localClient := k.client.(*k8s.Client)
+	localRelease, err := getRelease(localClient, k.params)
 	if err != nil {
 		k.Log("❌ Unable to find Helm release for the target cluster")
 		return err
 	}
-
-	localClient, remoteClient, err := k.getClientsForConnect()
+	remoteClients, err := k.getRemoteClients()
 	if err != nil {
 		return err
 	}
 
-	aiLocal, aiRemote, err := k.getAccessInfoForConnect(ctx, localClient, remoteClient)
+	localHelmValues := localRelease.Config
+	localOldClusters, err := getOldClusters(localHelmValues)
 	if err != nil {
 		return err
 	}
-
-	err = k.validateInfoForConnect(aiLocal, aiRemote)
+	aiLocal, err := k.getAccessInfoForConnect(ctx, localClient, k.params.SourceEndpoints)
 	if err != nil {
 		return err
 	}
 
-	// Validate that CA certificates match between the two clusters
-	match, err := k.validateCAMatch(aiLocal, aiRemote)
-	if err != nil {
-		return err
-	} else if !match {
-		k.Log("⚠️ Cilium CA certificates do not match between clusters. Multicluster features will be limited!")
-	}
+	localNewClusters := make([]map[string]interface{}, 0)
```

Comment:
Lots going on here... Not too keen about all this state and assignments. 
Is there a way we can clean this up a bit using functions to gather the necessary state?

Also depending on the connection type some of the modes may not require all that pre-computed state?
Lastly, breaking this up in sep func/methods would make testing this functionality easier.